src/compiler.ml

```diff
@@ -1319,12 +1414,33 @@ let query_preterm_of_ast ~depth macros state (loc, t) =
        ("Duplicate mode declaration for " ^ Symbols.show state name ^ " (also at "^
          Loc.show (snd (C.Map.find name map)) ^ ")")
 
-  let compile_mode (state, modes) { Ast.Mode.name; args; loc } =
-    let args = List.map to_mode args in
+  let to_mode = function Ast.Mode.Input -> Input | Output -> Output
```

Comment:
I really dont like `to_mode`. There should be a way to say that the Input (resp Output) in Ast is the same as the Input (resp Output) in Data, especially since both in Ast.Mode and Data, we define mode as `type mode = Util.arg_mode = Input | Output`